server/publications/collections/products.js

```diff
@@ -144,17 +144,37 @@ function extendSelectorWithVariants(collectionName, selector, productFilters, pr
   return newSelector;
 }
 
-function filterProducts(productFilters) {
```

Comment:
this file's diff looks more complex than it should. the summary of changes is that I pulled out the shop selection logic from both the "Products" and "Products/grid" methods into a common place. Doing so allowed for some code cleanup in each of the methods, which accounts for the reset of the diff